[RM Ch03] Storage Telemetry Definitions (#2377)
cntt-n/CNTT: doc/ref_model/chapters/chapter03.md

<a name="3.2.1.3"></a>
#### 3.2.1.3 Virtual Storage

> A workload can request different sizes and types of storage based on data longevity: persistent or ephemeral storage.

Reviewer: typo: 'size' instead of 'sizes'

Is the 'size' determined by 'data longevity'?

```suggestion
A workload can request storage based on data longevity (persistent or ephemeral storage), different types of storage (HDD, SSD, etc.) and storage size.
```

Author: LGTM, thanks

Persistent storage outlives the compute instance whereas ephemeral storage is linked to compute instance lifecycle.

> There are multiple storage performance attributes, such as latency, IO/s (Input/Output Operations per second, a.k.a. IOPS), and throughput. For example, a workload may require one of its storage devices to provide low latency, high IO/s and very large/huge storage size (terabytes of data).

Reviewer: are we defining the terms: **high** IO/s, **very large/huge** storage size

Author: It appears that the RM provides quantitative definitions for some of these performance attributes in https://cntt.readthedocs.io/en/latest/ref_model/chapters/chapter04.html#4.2.3  (as you pointed out today). The sentence that begins "For example, ..." is only modified to use consistent terms.

Low Latency storage is for workloads which have strong constraints on the time to access the storage.

> High IO/s oriented storage is for workloads requiring lots of read/write actions.

Reviewer: instead of "lots" can we have a quantitative value as in 4.2.3

Author: I suggest we reference https://cntt.readthedocs.io/en/latest/ref_model/chapters/chapter04.html#4.2.3 for the range of values. I don't know the syntax to do a forward reference though...

Reviewer: @acm If I were to use the url and attach it to the word 'reference" I would use the following syntax with the quotes:
       "["reference"]"(https://cntt.readthedocs.io/en/latest/ref_model/chapters/chapter04.html#4.2.3)

From Chapter 2 a reference to chapter 4 the relative reference would be "./chapter04.html#4.2.3"

Author: Thanks, I'll try a summary sentence to resolve your comments on qualitative terms. 
The last sentence of the paragraph is below, so I'll add it there.

> Large size storage is for workloads that need lots of volume without strong perfomance constraints.

Reviewer: Can we specify a value or range for "Large"

Author: I suggest we reference https://cntt.readthedocs.io/en/latest/ref_model/chapters/chapter04.html#4.2.3 for the range of values. I don't know the syntax to do a forward reference though...

Author: ```suggestion
Large size storage is for workloads that need lots of volume without strong performance constraints.
Note that approximate numeric ranges for the qualitative values used above are given in the 
"["Storage Extensions"]"(https://cntt.readthedocs.io/en/latest/ref_model/chapters/chapter04.html#4.2.3) section.
```

Storage resources have the following attributes, with metric definitions that support verification through passive measurements (telemetry) where appropriate:

| Attribute           | Description                                                                                    |
|---------------------|------------------------------------------------------------------------------------------------|
| `name`              | name of storage resources                                                                      |
| `data availibilty`  | persistent or ephemeral                                                                        |
|                     | Read and Write Latency, The average amount of time to perform a R/W operation, in milliseconds |

> | `performance`       | Read and Write IO/s, The average rate of performing R/W in IO operations per second            |

Reviewer: Can you move 'performance' to the row above as in the rendered table it appears Latency is part of 'data availability'

or use html tables

Author: I think I proposed the change below...

Author: Done below

> |                     | Read and Write Latency, The average amount of time to perform a R/W operation, in milliseconds |
> | `performance`       | Read and Write IO/s, The average rate of performing R/W in IO operations per second            |

Author: ```suggestion
| `performance`       | Read and Write Latency, The average amount of time to perform a R/W operation, in milliseconds |
|                     | Read and Write IO/s, The average rate of performing R/W in IO operations per second            |
```

<a name="3.2.1.3"></a>
#### 3.2.1.3 Virtual Storage

> A workload can request storage based on data availability (persistent or ephemeral storage), different types of storage (HDD, SSD, etc.) and storage size.

Reviewer 2: Availability is more about the resource being possible.  I think durability could be a better word for length of storage persistance.

Reviewer 2: Sorry, also the types of storage are not typically requested by workload.  Given something like openstack, you would request the storage from a specific pool by name.  While the pool name might be HDD, the actual workload can never know that is really is HDD and is definitely not allowed to know that level of detail about the underlying infrastructure.

Author: I agree we could replace **availability** with another term, and I think "retaining policy" may be approximately what we want to say.  IOW, data in ephemeral storage could last a very long time (such as the life of a VM), while data in persistent storage could be overwritten frequently - if its main purpose  is to recover state following a crash or other event. So, associating time with persistent and ephemeral might be right in many circumstances, but not all. 

Also, we already use *type* for  block, object or file storage.  
So, we won't mention HDD or SDD as attributes that workload can request.

Persistent storage outlives the compute instance whereas ephemeral storage is linked to compute instance lifecycle.

There are multiple storage performance attributes, such as latency, IO/s (Input/Output Operations per second, a.k.a. IOPS), and throughput. For example, a workload may require one of its storage devices to provide low latency, high IO/s and very large/huge storage size (terabytes of data).
Low Latency storage is for workloads which have strong constraints on the time to access the storage.
High IO/s oriented storage is for workloads requiring lots of read/write actions.
Large size storage is for workloads that need lots of volume without strong performance constraints.
Note that approximate numeric ranges for the qualitative values used above are given in the 

> "["Storage Extensions"]"(chapter04.html#4.2.3) section.

Reviewer: ```suggestion
[Storage Extensions](./chapter04.html#4.2.3) section.
```

Author: Thanks, WFM

> "["Storage Extensions"]"(https://cntt.readthedocs.io/en/latest/ref_model/chapters/chapter04.html#4.2.3) section.

Reviewer 3: ```suggestion
"["Storage Extensions"]"(chapter04.html#4.2.3) section.
```
Please use relative links to keep the links valid in multiple release branches.

Author: Thanks Greg0, better to reference within than external!
